plothist/hep_plotters.py

```diff
@@ -197,59 +150,105 @@ def plot_mc(
     mc_labels=None,
     mc_colors=None,
     signal_label="Signal",
+    fig=None,
+    ax=None,
     save_as=None,
     flatten_2d_hist=False,
+    stacked=True,
+    leg_ncol=1,
 ):
     """
-    Plot Monte Carlo (MC) simulations.
+    Plot MC simulations.
 
     Parameters
     ----------
     mc_hist_list : list of boost_histogram.Histogram
-        List of histograms representing the MC simulations.
+        The list of histograms for MC simulations.
     signal_hist : boost_histogram.Histogram, optional
-        Histogram representing the signal MC simulation, by default None.
+        The histogram for the signal. Default is None.
     xlabel : str, optional
-        Label for the x-axis, by default None.
+        The label for the x-axis. Default is None.
     ylabel : str, optional
-        Label for the y-axis, by default None.
+        The label for the y-axis. Default is None.
     mc_labels : list of str, optional
-        List of labels for the MC simulations, by default None.
+        The labels for the MC simulations. Default is None.
     mc_colors : list of str, optional
-        List of colors for the MC simulations, by default None.
+        The colors for the MC simulations. Default is None.
     signal_label : str, optional
-        Label for the signal histogram, by default "Signal".
-    save_as : str, optional
-        File path to save the figure, by default None.
+        The label for the signal. Default is "Signal".
+    fig : matplotlib.figure.Figure or None, optional
+        The Figure object to use for the plot. Create a new one if none is provided.
+    ax : matplotlib.axes.Axes or None, optional
+        The Axes object to use for the plot. Create a new one if none is provided.
+    save_as : str or None, optional
+        The file path to save the figure. Default is None.
     flatten_2d_hist : bool, optional
-        Whether to flatten 2D histograms, by default False.
+        If True, flatten 2D histograms to 1D before plotting. Default is False.
+    stacked : bool, optional
+        If True, stack the MC histograms. If False, plot them side by side. Default is True.
+    leg_ncol : int, optional
+        The number of columns for the legend. Default is 1.
 
     Returns
     -------
     fig : matplotlib.figure.Figure
-        The generated figure.
+        The Figure object containing the plot.
     ax : matplotlib.axes.Axes
-        Axes instance for the plot.
+        The Axes object containing the plot.
+
     """
 
+    if fig is None and ax is None:
+        fig, ax = plt.subplots()
+    elif (fig is None and ax) or (fig and ax is None):
```

Comment:
At this stage, we already know that at least one is not None, so just need to check if at least one is None, and raise the Error in this case